### PR TITLE
add runtime enable/disable for mkldnn matmul kernel

### PR DESCRIPTION
**Pull request by snadampal**
this control allows applications to disable the mkldnn matmul backend if the additional graph rewrite and onednn overhead is not acceptable for any matmul tensors.

use, export MKLDNN_MATMUL_ENABLE=0, to disable.

Fixes https://github.com/pytorch/pytorch/issues/96140

### CODE REVIEW

File: aten/src/ATen/native/LinearAlgebra.cpp [hunk 1]
```diff
@@ -1309,6 +1309,19 @@ Tensor outer(const Tensor& self, const Tensor& vec2) {
...
 
+#ifdef __aarch64__
+static inline bool is_mkldnn_matmul_enabled() {
+  static bool value = [&](const char* pt) {
+    if (pt != nullptr) {
+      return std::atoi(pt);
+    } else {
+      return 1;
+    }
+  }(std::getenv("MKLDNN_MATMUL_ENABLE"));
```
Comment (malfet) [inline aten/src/ATen/native/LinearAlgebra.cpp:1320]: Nits: Please use `TORCH_` prefix for all PyTorch env vars and use ternaries to make code a bit more compact.
```suggestion
  static bool value = [&]() {
    const char *ptr = std::getenv("TORCH_MKLDNN_MATMUL_ENABLE");
     return pt != nullptr ? std::atoi(pt) : 1;
  }();
```

**Review by malfet — APPROVED** [review]
Please consider using `constexpr` to avoid `#ifdef` guards in `addmm_impl_cpu_`. Also, we commonly prefix all PyTorch env vars with `TORCH_`

**Review by malfet — CHANGES_REQUESTED** [review]
Sorry, I didn't realize it targets 2.0 branch. For trunk this is totally fine.

**Review by malfet — APPROVED** [review]
LGTM, but please consider addressing nit and define `static constexpr bool is_mkldnn_matmul_enable() { return true;}` for non-aarch64 platforms to avoid extra `#ifdef` in `addmm_impl_cpu_`

**Comment by kit1980** [comment]
Hi @snadampal, I wonder if you're still interested in merging this?

**Comment by snadampal** [comment]
Hi @kit1980 , yes. btw, this PR was raised on 2.0 branch which is no longer relevant, but I have the below PR for the main branch which needs to be reviewed.
https://github.com/pytorch/pytorch/pull/96141

**Comment by kit1980** [comment]
Closing this in favor of https://github.com/pytorch/pytorch/pull/96141

**Pull request closed without merging**